### PR TITLE
Triangulation_2: Clarify what happens with degenerate constraints

### LINKED ISSUE #7336: Improve doc of CDT for border cases of constraint insertion
The doc should be updated to describe precisely, and the code updated to have a consistent behavior:
 
- `insert_constraints(points_begin, points_end, csts_begin, csts.end`: If `csts_begin==cst_end` not points is inserted but all the points are inserted even if `[cst_begin,cst_end[` does not use all the points. 
- `insert_constraint(Point a, Point b)` whats happening if `a==b`? Same questions for all variants with segment, pair,...

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Improve the documentation of  the constrained triangulation classes to make clear what happens 
- when degenerate constraints are inserted, 
- when a sequence of points and index pairs are inserted.

## Release Management

* Affected package(s): Triangulation_2
* Issue(s) solved (if any): fix #7336
* Link to compiled documentation (obligatory for small feature) [*wrong link name to be changed*](httpssss://wrong_URL_to_be_changed/Manual/Pkg)
* License and copyright ownership:  unchanged

### CODE REVIEW

File: Triangulation_2/doc/Triangulation_2/CGAL/Constrained_triangulation_plus_2.h [hunk 1]
```diff
@@ -22,11 +22,17 @@ point.
 
 
 The data structure maintains for each input constraint
-the sequence of vertices on this constraint. These vertices are
+the sequence of vertices on this constraint. Note that there is
+not a one-to-one correspondence between an input constraint and
+the sequence of vertices. These vertices are
 either vertices of the input constraint or intersection points.
+Also consecutive identical points in the input constraint
+result in a single vertex in the sequence if vertices on this
```
Comment (albert-github) [inline Triangulation_2/doc/Triangulation_2/CGAL/Constrained_triangulation_plus_2.h:30]: if -> of ?

File: Triangulation_2/doc/Triangulation_2/CGAL/Constrained_triangulation_plus_2.h [hunk 2]
```diff
@@ -264,10 +273,13 @@ inserts a polyline defined by the points in the range `[first,last)`
...
-In case the range is empty `Constraint_id()` is returned.
-In case all points are equal the point is inserted but no constraint,
-and `Constraint_id()` is returned.
+In case the range is empty a default constructed `Constraint_id` is returned.
+In case the range contains only one point or all points are equal the point is inserted but no constraint,
+and a default constructed `Constraint_id` is returned.
+
 \tparam PointIterator must be an `InputIterator` with the value type `Point`.
+
+\note
```
Comment (albert-github) [inline Triangulation_2/doc/Triangulation_2/CGAL/Constrained_triangulation_plus_2.h:282]: what should be noted ?

**Comment by sloriot** [comment]
Successfully tested in 5.6-Ic-208

**Review by sloriot — APPROVED** [review]

**Pull request merged**